src/bluetooth.py

```diff
@@ -0,0 +1,30 @@
+import bluetooth
+import pexpect
+import time
+
+def connect(targetName):
+    targetAddress = None
+
+    while targetAddress == None:
+        nearby_devices = bluetooth.discover_devices()
+        print(len(nearby_devices))
+        for i in nearby_devices:
+            print(bluetooth.lookup_name(i))
+            if targetName == bluetooth.lookup_name(i):
+                targetAddress = i
+
+        time.sleep(1)
+
+    print(targetAddress)
+
+
+    child = pexpect.spawn('bluetoothctl')
```

Comment:
This is exactly what I told you not to do. This is running the exact same commands that we are doing currently in bash. This just adds another layer of slow, unnecessary Python complexity. If this is the only way to do it, maybe this should just be done in bash.

Reply:
Yes, this is the only way I found that it could work. The L2CAP required a client and a server and I found another package called gatt but it would not work because it requires us to scan for an adapter.

Reply:
The specific Bluetooth packet protocol isn't what I'm concerned about. It doesn't matter if you're using L2CAP or GATT (you're scanning anyway) or RFCOMM or whatever. What matters is that you're still doing exactly what we're doing manually. 

> ```
>     child = pexpect.spawn('bluetoothctl')
>     child.sendline('connect ' + targetAddress)
>     child.sendline('exit')
>     child.interact()```

All this code is doing is literally sending "bluetoothctl" to the terminal and then piping "connect $ADDR" to it. I'm pretty sure this could be done with just one command in bash:
 ```
echo -e connect `5C:BA:37:26:6D:9A\nquit' | bluetoothctl
```

I don't think this Python solution is better than that one liner in bash. The Python is definitely slower. I gave you this task to see if there's a better way to do this. This task is crucial to the project.